elements/camera-capture/camera-capture.js

```diff
@@ -0,0 +1,320 @@
+    setTimeout(async () => {
+      await customElements.whenDefined('settings-pane');
+      settings.applyFromTrack(this.videoTrack);
+    }, 500);
+  }
+
+  async takePhoto() {
+    try {
+      const blob = await this.imageCapturer.takePhoto();
+      const img = await createImageBitmap(blob);
+
+      const canvas = this.shadowRoot.querySelector('#gallery');
+      canvas.width = getComputedStyle(canvas).width.split('px')[0];
+      canvas.height = getComputedStyle(canvas).height.split('px')[0];
+      let ratio = Math.max(canvas.width / img.width, canvas.height / img.height);
+      let x = (canvas.width - img.width * ratio) / 2;
+      let y = (canvas.height - img.height * ratio) / 2;
+      canvas.getContext('2d').clearRect(0, 0, canvas.width, canvas.height);
+      canvas.getContext('2d').drawImage(img, 0, 0, img.width, img.height,
+        x, y, img.width * ratio, img.height * ratio);
+    } catch(err) {
+      console.error("takePhoto() failed: ", err)
+    }
+  }
+
+  startCamera(target, constraints) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const stream = await navigator.mediaDevices.getUserMedia({
+          video: constraints,
+          audio: false
+        });
+
+        this.video = target;
+        this.stream = stream;
+
+        target.srcObject = stream;
+        target.addEventListener('canplay', resolve, { once: true });
+        target.play();
+      } catch(err) {
+        reject(err);
+      }
+    });
+  }
+
+  stopCamera() {
+    if (this.video) {
+      this.video.pause();
+      this.video.srcObject = null;
+    }
+    if (this.stream) {
+      this.stream.getVideoTracks()[0].stop();
+    }
+  }
+
+  getFacingMode(device) {
+    if (device.facingMode == "environment"
+      || device.label.indexOf("facing back") >= 0) {
+      return "environment";
+    }
+    // We assume by default that cameras are user facing
+    // which is mostly the case for desktop.
+    return "user";
+  }
+
+  async firstUpdated() {
+    this.constraints = {};
+
+    const devices = await navigator.mediaDevices.enumerateDevices();
+
+    this.cameras = [];
+    this.selectedCamera = 0;
+
+    devices.forEach(device => {
+      if (device.kind == 'videoinput') {
+        if (this.getFacingMode(device) == "user") {
+          this.cameras.push(device);
+        } else {
+          this.cameras.unshift(device);
+        }
+      }
+    });
+
+    // Android bug, doesn't work with DevTools emulation.
+    if (navigator.userAgent.includes("Android") &&
+      screen.orientation.type.includes("portrait")) {
+      this.constraints.width = Math.ceil(visualViewport.height);
+      this.constraints.height = Math.ceil(visualViewport.width);
+    } else {
+      this.constraints.width = Math.ceil(visualViewport.width);
+      this.constraints.height = Math.ceil(visualViewport.height);
+    }
+
+    // Disable facingModeButton if there is no environment or user mode.
```

Comment:
Let's elaborate a bit more on the comment, as we discussed ? Do we want to treat that icon as a camera switcher (say 2 user facing mode cameras on a laptop, the inbuilt and an external one), or a facing mode switcher like say on Android, (front/user and back/environment)

Reply:
Switching camera on a mobile phone with just two cameras is the exact same behavior as switching between front and back